### ISSUE TITLE
hcp_vault_secrets_app description parameter is listed as optional but resource creation fails if not provided

### ISSUE THREAD

**Issue by kotom541** [hashicorp/terraform-provider-hcp#655]
Terraform version: 1.6.2
HCP provider version: v0.74.1

### Affected Resource(s)

resource hcp_vault_secrets_app

### Terraform Configuration Files

```hcl
terraform {
  cloud {
    organization = "PRIVATE"

    workspaces {
      name = "PRIVATE"
    }
  }
  required_providers {
    hcp = {
      source  = "hashicorp/hcp"
      version = ">=0.74.1"
    }
  }
}

provider "hcp" {
  project_id = "PRIVATE"
  client_id     = var.hcp_client_id
  client_secret = var.hcp_client_secret
}
```

```hcl
variable "hcp_client_id" {
    type = string
}

variable "hcp_client_secret" {
    type = string
    sensitive = true
}
```

```hcl
resource "hcp_vault_secrets_app" "demo" {
  app_name = "hcp-bug-demo"
}
```

### Debug Output

│ Error: Provider produced inconsistent result after apply
│
│ When applying changes to hcp_vault_secrets_app.demo, provider
│ "provider[\"registry.terraform.io/hashicorp/hcp\"]" produced an unexpected
│ new value: .description: was null, but now cty.StringVal("").
│
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker

### Steps to Reproduce

1. `terraform apply`

### Expected Behavior

hcp_vault_secrets_app resource should be created with blank description.

### Actual Behavior

hcp_vault_secrets_app resource failed to be created.

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by maxcoulombe** [comment]
Hey, thanks for reporting this issue. It should be addressed in #1118 among a number of additional fixes for the HVS App resource!

**Comment by maxcoulombe** [comment]
All right the PR is merged, the description should be optional as expected. A handful of other fixes were also listed in the PR's description. Let us know if there is anything else!
